src/Parser/VectorTileParser.js

```diff
@@ -0,0 +1,85 @@
+    };
+
+    layers.forEach((layer_id) => {
+        const l = vectorTile.layers[layer_id];
+
+        for (let i = 0; i < l.length; i++) {
+            let feature;
+            // We need to move from TMS to Google/Bing/OSM coordinates
+            // https://alastaira.wordpress.com/2011/07/06/converting-tms-tile-coordinates-to-googlebingosm-tile-coordinates/
+            // Only if the layer.origin is top
+            if (options.origin == 'top') {
+                feature = l.feature(i).toGeoJSON(options.coords.col, options.coords.row, options.coords.zoom);
+            } else {
+                const y = 1 << options.coords.zoom;
+                feature = l.feature(i).toGeoJSON(options.coords.col, y - options.coords.row - 1, options.coords.zoom);
+            }
+            if (layers.length > 1) {
+                feature.properties.vt_layer = layer_id;
+            }
+
+            geojson.features.push(feature);
+        }
+    });
+
+    return GeoJsonParser.parse(geojson, {
+        json: true,
+        crsOut: options.extent.crs(),
+        filteringExtent: options.filteringExtent,
+        filter: options.filter,
+        buildExtent: true,
+    });
+}
+
+/**
+ * @module VectorTileParser
+ */
+export default {
+    /**
+     * Parse a vector tile file and return a [Feature]{@link module:GeoJsonParser.Feature}
+     * or an array of Features. While multiple formats of vector tile are
+     * available, the only one supported for the moment is the
+     * [Mapbox Vector Tile]{@link https://www.mapbox.com/vector-tiles/specification/}.
+     *
+     * @param {ArrayBuffer} file - The vector tile file to parse.
+     * @param {Object=} options - Options controlling the parsing.
+     * @param {string=} options.format - 'application/x-protobuf;type=mapbox-vector'
+     * by default, which corresponds to a pbf file and a mapbox vector tile.
+     * @param {Extent=} options.extent - The Extent to convert the input coordinates to.
+     * @param {Extent=} options.filteringExtent - Optional filter to reject features
+     * outside of this extent.
+     * @param {function=} options.filter - Filter function to remove features.
+     * @param {string=} options.origin - This option is to be set to the correct
+     * value, bottom or top (default being bottom), if the computation of the
+     * coordinates needs to be inverted to same scheme as OSM, Google Maps or
+     * other system. See [this link]{@link https://alastaira.wordpress.com/2011/07/06/converting-tms-tile-coordinates-to-googlebingosm-tile-coordinates} for more informations.
+     * @param {Extent=} options.coords - Coordinates of the layer.
+     *
+     * @return {Promise} A Promise resolving with a Feature or an array a
+     * Features.
+     */
+    parse(file, options = {}) {
+        switch (options.format) {
+            case 'application/x-protobuf;type=mapbox-vector':
+            default:
+                return Promise.resolve(readPBF(file, options));
+        }
```

Comment:
Should not you `Promise.reject` the default case ?

Reply:
I was on the assumption that if someone uses this parser, the file is always a vector tile, whatever the format, with the default case being being this one. So no rejection, as we trust the user to know what he does.

Reply:
If you trust the user, then there is no use to switch on the format (maybe you have an extension in mind, but for now the switch is useless, just return the resolve directly).
If you want to advertise the mimetype, I propose in #723 to make it a field of the parsers.

Reply:
Yes, that's why I'm eager to see the parsers coming here :grinning: